src/main/java/com/dabsquared/gitlabjenkins/trigger/handler/push/PushHookTriggerHandlerFactory.java

```diff
@@ -12,20 +13,20 @@
 
     private PushHookTriggerHandlerFactory() {}
 
-    public static PushHookTriggerHandler newPushHookTriggerHandler(boolean triggerOnPush, TriggerOpenMergeRequest triggerOpenMergeRequestOnPush) {
+    public static PushHookTriggerHandler newPushHookTriggerHandler(GitLabPluginMode gitLabPluginMode, boolean triggerOnPush, TriggerOpenMergeRequest triggerOpenMergeRequestOnPush) {
         if (triggerOnPush || triggerOpenMergeRequestOnPush == TriggerOpenMergeRequest.both) {
-            return new PushHookTriggerHandlerList(retrieveHandlers(triggerOnPush, triggerOpenMergeRequestOnPush));
+            return new PushHookTriggerHandlerList(retrieveHandlers(gitLabPluginMode, triggerOnPush, triggerOpenMergeRequestOnPush));
         } else {
             return new NopPushHookTriggerHandler();
         }
     }
 
-    private static List<PushHookTriggerHandler> retrieveHandlers(boolean triggerOnPush, TriggerOpenMergeRequest triggerOpenMergeRequestOnPush) {
+    private static List<PushHookTriggerHandler> retrieveHandlers(GitLabPluginMode gitLabPluginMode, boolean triggerOnPush, TriggerOpenMergeRequest triggerOpenMergeRequestOnPush) {
         List<PushHookTriggerHandler> result = new ArrayList<PushHookTriggerHandler>();
         if (triggerOnPush) {
             result.add(new PushHookTriggerHandlerImpl());
         }
-        if (triggerOpenMergeRequestOnPush == TriggerOpenMergeRequest.both) {
+        if (gitLabPluginMode == GitLabPluginMode.LEGACY && triggerOpenMergeRequestOnPush == TriggerOpenMergeRequest.both) {
```

Comment:
Nope. I just tested this with GitLab 8.7:
1. I configured the job in Jenkins as you described in the README
2. I configured my GitLab repo to send MR events to the Jenkins
3. I created a MR for a test branch into the master -> a new build was triggerd
4. I pushed some changes to the master (target branch) -> no build was triggered as no event was sent by GitLab